Source/WebInspectorUI/UserInterface/Controllers/ConsoleManager.js

```diff
@@ -176,24 +178,34 @@ WI.ConsoleManager = class ConsoleManager extends WI.Object
 
         WI.ConsoleCommandResultMessage.clearMaximumSavedResultIndex();
 
-        if (this._clearMessagesRequested) {
-            // Frontend requested "clear console" and Backend successfully completed the request.
-            this._clearMessagesRequested = false;
-
-            this._warningCount = 0;
-            this._errorCount = 0;
-            this._issues = [];
-
-            this._lastMessageLevel = null;
+        // COMPATIBILITY (iOS 16.0, macOS 13.0): `Console.ClearReason` did not exist.
+        if (!InspectorBackend.Enum.Console?.ClearReason) {
+            if (this._clearMessagesRequested) {
+                // Frontend requested "clear console" and Backend successfully completed the request.
+                this._clearMessagesRequested = false;
+                this._internalClearMessages();
+                return;
+            }
 
-            this.dispatchEventToListeners(WI.ConsoleManager.Event.Cleared);
-        } else {
             // Received an unrequested clear console event.
             // This could be for a navigation or other reasons (like console.clear()).
             // If this was a reload, we may not want to dispatch WI.ConsoleManager.Event.Cleared.
             // To detect if this is a reload we wait a turn and check if there was a main resource change reload.
             setTimeout(this._delayedMessagesCleared.bind(this), 0);
         }
+
+        switch (reason) {
+        case WI.ConsoleManager.ClearReason.ConsoleClearAPI:
+            this._internalClearMessages();
+            break;
+        case WI.ConsoleManager.ClearReason.MainFrameNavigation:
+            if (!WI.settings.clearLogOnNavigate.value)
+                break;
```

Comment:
what about all the other stuff that's inside `_mainResourceDidChange`?

NIT: instead of a second `break` we could just
```
if (WI.settings.clearLogOnNavigate.value)
    this._clearMessages();
break;
```